packages/tldraw/src/lib/shapes/shared/useImageOrVideoAsset.ts

```diff
@@ -108,10 +105,19 @@ export function useImageOrVideoAsset({ shapeId, assetId, width }: UseImageOrVide
 
 			// If we already resolved the URL, debounce fetching potentially multiple image variations.
 			if (didAlreadyResolve.current) {
-				resolveAssetUrlDebounced(editor, assetId, screenScale, exportInfo, (url) =>
-					resolve(asset, url)
-				)
-				cancelDebounceFn = resolveAssetUrlDebounced.cancel // cancel the debounce when the hook unmounts
+				let tick = 0
+
+				const resolveAssetAfterAWhile = () => {
+					tick++
+					if (tick > 32) {
```

Comment:
why this change along with this PR? this makes the loading of the resized image asset delayed by about ~500ms when zooming is finished. is that the intent?

Reply:
On the torture test that Mitja sent to me, the debounce was actually a performance hit (setting and removing the global timers). We could pull that from this PR and look at it separately, but my thinking here is that we have a tick going anyway, so we could use that.

Reply:
(my bad, btw, the original one also does 500ms :doh:)